lib/form_data.js

```diff
@@ -31,6 +31,7 @@ function FormData(options) {
   this._overheadLength = 0;
   this._valueLength = 0;
   this._valuesToMeasure = [];
+  this._entryList = [];
```

Comment:
I am confused with this.

You use an array as the data structure but then you use it as an object

Reply:
My bad ! 
It works because Javascript Array are objects, and thus it become attribute of an object, I will change this to a map ;)

Reply:
@jdeniau actually I would use a list for this and create a list of objects.

I know that from data structure makes sense but following the spec I believe that `append` do not remove the previous one.

From documentation


> Each FormData object has an associated entry list (a list of entries). It is initially the empty list.
> An entry consists of a name and a value.


So maybe is better to have a  list of objects

```
[
 { name: 'hello', value: 'world'}
]
```

I am trying to finish the spec and probably maybe would be better if you change the current implementation

So far,

```
function createEntry(name, value, filename) {
  // TODO: handle Blob and File
  // Related to https://xhr.spec.whatwg.org/#concept-formdata-entry

  // Case 1:
  // If value is a Blob object and not a File object, then set value to a new
  // File object, representing the same bytes, whose name attribute value is
  // "blob".

  // Case 2:
  // If value is (now) a File object and filename is given, then set value to a
  // new File object, representing the same bytes, whose name attribute value
  // is filename.

  var entry = {
    name: name,
    value: value,
  };

  return entry;
}
```

I created this functions because it seems that we will need to do some complex stuff with the entries  https://xhr.spec.whatwg.org/#concept-formdata-entry

So basically for now you could do `this._entries.push( createEntry(name, value, filename  );`